src/jit/optcse.cpp

```diff
@@ -788,9 +798,22 @@ void Compiler::optCseUpdateCheckedBoundMap(GenTree* compare)
  *
  *  Compute each blocks bbCseGen
  *  This is the bitset that represents the CSEs that are generated within the block
+ *  Also initialize bbCseIn, bbCseOut and bbCseGen sets for all blocks
  */
 void Compiler::optValnumCSE_InitDataFlow()
 {
+    // Init traits and full/empty bitvectors.  This will be used to track the
+    // individual cse indexes.
+
+    // two bits, plus one extra bit is allocated per CSEcandidate to compute CSE liveness
```

Comment:
I would reword this as:
```
// We allocate two bits per CSE candidate, plus an extra bit to handle the initial unvisited case.
// The first bit tracks ...
```
And somewhere it would be good to describe how that extra bit is used to handle the unvisited case.